lib/config/defaults.js

```diff
@@ -364,16 +367,56 @@ const applyModuleDefaults = (
 			}
 		];
 		if (mjs) {
-			rules.push({
-				test: /\.mjs$/i,
+			const esm = {
 				type: "javascript/esm",
 				resolve: {
-					mainFields: webTarget ? ["browser", "main"] : ["main"]
+					byDependency: {
+						esm: {
+							fullySpecified: true
+						},
+						wasm: {
+							fullySpecified: true
+						}
+					}
 				}
-			});
+			};
+			const commonjs = {
+				type: "javascript/dynamic"
+			};
+			rules.push(
+				{
+					test: /\.mjs$/i,
+					...esm
+				},
+				{
+					test: /\.js$/i,
+					descriptionData: {
+						type: "module"
+					},
+					...esm
```

Comment:
So to sum up, to enable "javascript/esm" I need add `experiments: {mjs: true}` and `type: module` in description file?

Reply:
maybe add another experiment for this?

Reply:
`*.mjs` and `type: "module"` is both part of the node.js behavior.

I actually think about removing the experiment and enable it by default. I'm a bit worries about node.js ESM support is still flagged as experimental...